broker-core/src/main/java/org/apache/qpid/server/transport/NonBlockingConnection.java

```diff
@@ -691,4 +693,12 @@ public String getSelectedHost()
     {
         return _selectedHost;
     }
+
+    public boolean detectConnectionLooping() { return _port.getContextValue(Boolean.class, AmqpPort.PORT_DETECT_CONNECTION_LOOPING); }
+
+    public boolean isLooping() { return _loopingCounter.get() > _port.getContextValue(Integer.class, AmqpPort.PORT_CONNECTION_LOOPING_DETECTION_THRESHOLD); }
```

Comment:
It needs to be moved into `NonBlockingConnectionTLSDelegate` as it is only used there.

A field can be added into `NonBlockingConnectionTLSDelegate`  to hold a value, as evaluation of context variable is expensive operation.